duecredit/collector.py

```diff
@@ -253,7 +268,8 @@ def cite(self, entry, **kwargs):
             citation = self.citations[citation_key]
         except KeyError:
             self.citations[citation_key] = citation = Citation(entry_, **kwargs)
-        assert(citation.key == citation_key)
+        assert type(citation) is Citation
```

Comment:
Unless you want to prohibit subclassing, this would be better written as `assert isinstance(citation, Citation)`.